osimpipeline/study.py

```diff
@@ -17,25 +17,67 @@ def __init__(self, trial, num, gait_landmarks, metadata=None):
         self.num = num
         self.name = 'cycle%02i' % num
         self.metadata = metadata
-        self.rel_path = os.path.join(trial.rel_path, self.name)
         self.id = '_'.join([trial.id, self.name])
+        self.start = gait_landmarks.cycle_start
+        self.end = gait_landmarks.cycle_end
+
+        self.tasks = list()
+
+    def add_task(self, cls, trial, *args, **kwargs):
+        """Add a TrialTask for this cycle.
+            
+           TrialTasks for cycles can only be created
+           by Trial objects. 
+        """
+        kwargs['cycle']=self
+        task = cls(trial, *args, **kwargs)
+        self.tasks.append(task)
+        return task
 
 class Trial(object):
+    """Trials may have multiple cycles. If a condition has only one
+    trial, you can omit the unnecessary trial
+    directory using `omit_trial_dir`.
+
+    Examples
+    --------
+
+    ```
+    new_trial = cond.add_trial(1)
+
+    ```
+
+    ```
+    new_trial = cond.add_trial(1, start_time=0.0, end_time=0.5)
+
+    ```
+
+    ```
+    new_trial = cond.add_trial(1, right_strikes=[45.90,47.00,52.00],
+                                  right_toeoffs=[46.64,49.00],  
+                                  left_strikes=[46.45,50.00],
+                                  left_toeoffs=[47.19,51.00],
+                                  omit_trial_dir=True)
+    ```
+    """
     def __init__(self, condition, num, metadata=None,
-            omit_trial_dir=False,
+            omit_trial_dir=False, primary_leg=False,
+            start_time=None, end_time=None,
+            right_strikes=None, right_toeoffs=None,
+            left_strikes=None, left_toeoffs=None,
```

Comment:
I like that idea. What about `gait_events` or even just `events` instead of `strikes_and_toeoffs`?